File: Data Structures & Algorithms/binary-tree-diameter/submission-0.cpp
class Solution {
public:
    int res = 0;
    int height(TreeNode* node) {
        if (!node) return 0;
        int left = height(node->left);
        int right = height(node->right);
        res = max(res, left + right);
        return 1 + max(left, right);
    }

    int diameterOfBinaryTree(TreeNode* root) {
        height(root);
        return res;
    }
};

File: Data Structures & Algorithms/max-water-container/submission-0.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:
        left = 0
        right = len(heights) - 1
        maxArea = 0

        while left < right:
            if heights[left] <= heights[right]:
                maxArea = max((right - left) * min(heights[left], heights[right]), maxArea)
                left += 1
            elif heights[right] < heights[left]:
                maxArea = max((right - left) * min(heights[left], heights[right]), maxArea)
                right -= 1


        return maxArea